spec/javascript/availability/components/ill_link.jsx.spec.js

```diff
@@ -0,0 +1,194 @@
+      title_245ab_tsim: 'book title',
+      pub_date_illiad_ssm: 2021,
+      isbn_ssm: '1234'
+    });
+  });
+
+  const {getByRole, container} = render(
+    <IllLink holding={{locationID: 'BINDERY', callNumber: '123'}} />
+  );
+
+  expect(getByRole('link')).toHaveAttribute('href', '#');
+  expect(container.getElementsByClassName('spinner-border').length).toBe(1);
+
+  await waitFor(() => expect(container.getElementsByClassName('spinner-border').length).toBe(0));
+
+  expect(getByRole('link')).toHaveTextContent('Copy unavailable, request via Interlibrary Loan');
+
+  expect(getByRole('link')).toHaveAttribute('href',
+    'https://psu-illiad-oclc-org.ezaccess.libraries.psu.edu/illiad/upm/illiad.dll/OpenURL?' +
+    'Action=10&Form=30&isbn=1234&title=book%20title&callno=123' +
+    '&rfr_id=info%3Asid%2Fcatalog.libraries.psu.edu&date=2021');
+});
+
+test('renders a link with no publication date', async () => {
+  $.get = jest.fn().mockImplementation(() => {
+    return Promise.resolve({
+      title_245ab_tsim: 'book title',
+      author_tsim: 'author name',
+      isbn_ssm: '1234'
+    });
+  });
+
+  const {getByRole, container} = render(
+    <IllLink holding={{locationID: 'BINDERY', callNumber: '123'}} />
+  );
+
+  expect(getByRole('link')).toHaveAttribute('href', '#');
+  expect(container.getElementsByClassName('spinner-border').length).toBe(1);
+
+  await waitFor(() => expect(container.getElementsByClassName('spinner-border').length).toBe(0));
+
+  expect(getByRole('link')).toHaveTextContent('Copy unavailable, request via Interlibrary Loan');
+
+  expect(getByRole('link')).toHaveAttribute('href',
+    'https://psu-illiad-oclc-org.ezaccess.libraries.psu.edu/illiad/upm/illiad.dll/OpenURL?' +
+    'Action=10&Form=30&isbn=1234&title=book%20title&callno=123' +
+    '&rfr_id=info%3Asid%2Fcatalog.libraries.psu.edu&aulast=author%20name');
+});
+
+test('renders a link with no ISBN', async () => {
+  $.get = jest.fn().mockImplementation(() => {
+    return Promise.resolve({
+      title_245ab_tsim: 'book title',
+      author_tsim: 'author name',
+      pub_date_illiad_ssm: 2021
+    });
+  });
+
+  const {getByRole, container} = render(
+    <IllLink holding={{locationID: 'BINDERY', callNumber: '123'}} />
+  );
+
+  expect(getByRole('link')).toHaveAttribute('href', '#');
+  expect(container.getElementsByClassName('spinner-border').length).toBe(1);
+
+  await waitFor(() => expect(container.getElementsByClassName('spinner-border').length).toBe(0));
+
+  expect(getByRole('link')).toHaveTextContent('Copy unavailable, request via Interlibrary Loan');
+
+  expect(getByRole('link')).toHaveAttribute('href',
+    'https://psu-illiad-oclc-org.ezaccess.libraries.psu.edu/illiad/upm/illiad.dll/OpenURL?' +
+    'Action=10&Form=30&isbn=&title=book%20title&callno=123' +
+    '&rfr_id=info%3Asid%2Fcatalog.libraries.psu.edu&aulast=author%20name&date=2021');
+});
+
+test('renders a link when the response comes back with no data', async () => {
```

Comment:
@banukutlu - what should actually happen here? Right now, if the AJAX call responds with no data, it generates and shows this seemingly incomplete link:

`https://psu-illiad-oclc-org.ezaccess.libraries.psu.edu/illiad/upm/illiad.dll/OpenURL?Action=10`

I'd think it might be better to just hide/remove the link altogether if there is no data. But maybe this should not be changed in this PR, because the test is at least testing what is actually happening right now.

Reply:
hmm yeah that seems like a bug, we should add an issue for this.

Reply:
Opened a ticket for this: #848. Will remove the implementation for this test and mark it as pending/todo.